jaxopt/_src/osqp.py

```diff
@@ -599,3 +597,187 @@ def __post_init__(self):
+  """Convert instance of ``OSQP`` (which is ``CVXPY_QP`` compatible) to BoxOSQP."""
+
+  @staticmethod
+  def transform_matvec(matvec_Q, matvec_A, matvec_G):
+    """Return matvec_Q, matvec_A of BoxOSQP from the matvec_Q, matvec_A, matvec_G of OSQP."""
+    if matvec_A is None and matvec_G is None:
+      matvec_A_box = None
+    elif matvec_A is None:
+      def matvec_A_box(params, x):
+        A, params_G = params
+        return DenseLinearOperator(A)(x), matvec_G(params_G, x)
+    elif matvec_H is None:
+      def matvec_A_box(params, x):
+        params_A, G = params
+        return matvec_A(params_A, x), DenseLinearOperator(G)(x)
+    else:
+      def matvec_A_box(params, x):
+        params_A, params_H = params
+        return matvec_A(params_A, x), matvec_G(params_G, x)
+    return matvec_Q, matvec_A_box
+  
+  @staticmethod
+  def fit_transform(params: Optional[jnp.ndarray],
+                    params_obj: Tuple[Optional[Any], Any],
+                    params_eq: Optional[Tuple[Any,Any]],
+                    params_ineq: Optional[Tuple[Any,Any]]):
+    """Transform parameters of run().
+    
+    Supports only matrices and arrays."""
+    Q, c = params_obj
+
+    eq_ineq_size = [None, None]
+    A_box, l, u, y = [], [], [], []
+    if params_eq is not None:
+      A, b = params_eq
+      l.append(b)
+      u.append(b)
+      A_box.append(A)
+      eq_ineq_size[0] = A.shape[0]
+      if params is not None:
+        y.append(params.dual_eq)
+
+    if params_ineq is not None:
+      G, h = params_ineq
+      l.append(-jnp.inf * jnp.ones_like(h))
+      u.append(h)
+      A_box.append(G)
+      eq_ineq_size[1] = G.shape[0]
+      if params is not None:
+        y.append(params.dual_ineq)
+
+    # TODO(lbethune): does it make sense to support QP without constraints ?
+    # should the user switch to another solver instead ? (e.g order 2 methods... )
+    if not A_box:
+      raise ValueError("At least one of params_eq or params_ineq must be not None.")
+
+    l = jnp.concatenate(l, axis=0)
+    u = jnp.concatenate(u, axis=0)
+    A_box = jnp.concatenate(A_box, axis=0)
+
+    if params is not None:
+      x, z = params.primal, A_box @ params.primal
+      y = jnp.concatenate(y, axis=0)
+      params = BoxOSQP._get_full_KKT_solution((x, z), y)
+
+    hyper_params = dict(params_obj=(Q, c), params_eq=A_box, params_ineq=(l, u))
+    return params, hyper_params, eq_ineq_size
+  
+  @staticmethod
+  def inverse_transform(kkt_solution, eq_ineq_size):
+    """Inverse transform the KKT solution returned by run()"""
+    box_primal, box_dual_eq, (box_mu, _) = kkt_solution
+    x, _ = box_primal
+
+    dual_eq, dual_ineq = None, None
+    if eq_ineq_size[0] is not None:
+      dual_eq = box_dual_eq[:eq_ineq_size[0]]
+    if eq_ineq_size[1] is not None:
+      dual_ineq = box_mu[-eq_ineq_size[1]:]
+    
+    return base.KKTSolution(x, dual_eq, dual_ineq)
+
+
+class OSQP(base.Solver):
+  """OSQP solver written in Jax with same interface than CVXPY_QP.
```

Comment:
```suggestion
  """OSQP solver for general quadratic programming.
```
(remember that this is the line appearing in the "API at a glance" section)